### PR TITLE
Switch to onyen instead of email for new users

### PR DESCRIPTION
**Pull request by lfarrell**
https://jira.lib.unc.edu/browse/HYC-301

### CODE REVIEW

File: app/controllers/accounts_controller.rb [hunk 1]
```diff
@@ -17,15 +17,15 @@ def create
...
   def ensure_admin!
     authorize! :read, :admin_dashboard
   end
 
-  def create_user(email)
-    onyen = email.split('@').first
+  def create_user(onyen)
+    email = "#{onyen}@ad.unc.edu"
     if User.where(uid: onyen).blank?
       user = User.where(provider: 'shibboleth', uid: onyen, email: email).first_or_create
```
Comment (smithjp) [inline app/controllers/accounts_controller.rb]: This query needs to be changed to `User.where(uid: onyen).first_or_create(provider: 'shibboleth', email: email)`
Also the query on line 32 of the user model needs to be changed similarly. 

I think those might be the only two queries that need to change.

**Comment by bbpennel** [comment]
Have we tested this on QA yet?

**Comment by lfarrell** [comment]
Not that I'm aware of

**Comment by bbpennel** [comment]
I think we should try it there when it is available for a deploy, since we can't really test this locally.

**Comment by smithjp** [comment]
We just tested this on QA, and it seems to be working.

**Pull request merged**